Added validation for generated files

Makdaam opened the pull request:
This should missing the go generate step when it comes to generated Go code and deploy/*.json files.

jim-minter commented:
I was hoping to avoid this heavyweight kind of thing - currently in ci.yml we do:
```
          set -x
          make test-go
          [[ -z "$(git status -s)" ]]
```

That is intended to error out if git status indicates a file has changed.  It was working before we moved to Azure Pipelines but something has broken along the way - please could you see what that is?  Thanks!

Makdaam commented:
@jim-minter fixed the ci.yaml definition.
Rationale note for the future:
Not doing "set -xe" because we want to process the coverage reports even if the tests fail.

Makdaam commented:
Had to use set -xe after all. We would have missed make test-go returning non-0 exit code.
Moved the report preparation to a separate task. 
@julienstroheker could you take a look if I implemented the reports part correctly?

jim-minter commented:
Lgtm. @julienstroheker ?

julienstroheker commented on .pipelines/ci.yml:
```diff
@@ -46,14 +46,20 @@ jobs:
         workingDirectory: "${{ variables.modulePath }}"
-        displayName: "🧪Run Golang Unit Tests"
+        displayName: "⚙️ Process Reports"
+        condition: succeededOrFailed()
```
I guess we should also do the same of the `PublishTestResults` task ?

julienstroheker replied:
mb, its there !
